src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -203,6 +203,9 @@ uint256 AsyncRPCOperation_sendmany::main_impl() {
     // creating dust change amounts.
     CAmount dustThreshold{DefaultDustThreshold()};
 
+    // Required for FindSpendableInputs
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
     // Find spendable inputs, and select a minimal set of them that
     // can supply the required target amount.
     auto spendable = pwalletMain->FindSpendableInputs(ztxoSelector_, allowTransparentCoinbase, mindepth_);
```

Comment:
This lock on `cs_main` is being held for the entire duration of the async call, causing it to be synchronous with the chain updates and any other RPC calls that query chain state.
```suggestion
    // Find spendable inputs, and select a minimal set of them that
    // can supply the required target amount.
    SpendableInputs spendable;
    {
        LOCK2(cs_main, pwalletMain->cs_wallet);
        spendable = pwalletMain->FindSpendableInputs(ztxoSelector_, allowTransparentCoinbase, mindepth_);
    }
```